core/dbt/include/global_project/macros/adapters/metadata.sql

```diff
@@ -1,3 +1,19 @@
+{% macro get_catalog_relations(information_schema, relations_by_schema) -%}
```

Comment:
I like this approach for opting-in to the new approach. But if we update the macro name in `adapter/base/relation.py`, does it matter? Do we get what we're looking for? I'm not sure how often adapter maintainers actually overwrite that, so overwriting the name in `adapter/base/relation.py` may have the same effect as just updating the contents of the `get_catalog` macro.

Also, if we're breaking this out as a separate macro and making relations_by_schema a `Dict[str, Optional[List[BaseRelation]]]` (emphasis on `BaseRelation` vs `str`), then couldn't we make it just `List[BaseRelation]]`? `BaseRelation` has a `schema` property already, so doing the work of organizing by schema may be redundant.